Project/Source/Components/ComponentObstacle.cpp

```diff
@@ -29,7 +29,7 @@ void ComponentObstacle::Update() {
 	if (!newPosition.Equals(currentPosition) || !newRotation.Equals(currentRotation)) {
 		currentPosition = newPosition;
 		currentRotation = newRotation;
-		AddObstacle();
+		if (IsActive()) AddObstacle();
```

Comment:
`else { RemoveObstacle(); }` ?

Reply:
I don't think it's needed because if the component is initally disabled, the obstalce won't exist, and if it's disabled from Script or Editor,it will call the OnDisable that already deletes the obstacle and they will never (or at least should never) get to the Update method